cirq/google/calibration/workflow.py

```diff
@@ -14,24 +14,50 @@
 from typing import Callable, Dict, List, Optional, Sequence, Tuple, Union, cast
 
 import dataclasses
+from itertools import zip_longest
 
 
 from cirq.circuits import Circuit
-from cirq.ops import FSimGate, Gate, GateOperation, MeasurementGate, Moment, Qid, SingleQubitGate
+from cirq.ops import (
+    FSimGate,
+    Gate,
+    GateOperation,
+    MeasurementGate,
+    Moment,
+    Qid,
+    SingleQubitGate,
+    rz,
+)
 from cirq.google.calibration.engine_simulator import PhasedFSimEngineSimulator
 from cirq.google.calibration.phased_fsim import (
     FloquetPhasedFSimCalibrationOptions,
     FloquetPhasedFSimCalibrationRequest,
     IncompatibleMomentError,
     PhasedFSimCalibrationRequest,
     PhasedFSimCalibrationResult,
+    PhasedFSimCharacterization,
     WITHOUT_CHI_FLOQUET_PHASED_FSIM_CHARACTERIZATION,
+    ZETA_GAMMA_FLOQUET_PHASED_FSIM_CHARACTERIZATION,
     try_convert_sqrt_iswap_to_fsim,
 )
 from cirq.google.engine import Engine
 from cirq.google.serializable_gate_set import SerializableGateSet
 
 
+@dataclasses.dataclass(frozen=True)
+class CircuitCalibration:
```

Comment:
I could use some better name suggestions for this

Reply:
Do you expect users to be constructing this object? Maybe it should be private (i.e. starting with an underscore). As for names: it seems important that we're mapping from the contained circuit to these characterization records, so maybe something like `CircuitRequestResponseMapper` or `CircuitToRecord`. I made up the term "record" meaning "either a request or response". Maybe there's a better word than that

Reply:
This should be actually CircuitWithCharacterizationIndex or CircuitWithCalibrationIndex in the future. I made it to short CircuitWithCalibration for now. I want to schedule meeting with some of you to discuss all the naming issues soon.

Reply:
As for user construction, this object is a public interface of a method and shouldn't be public. I can imagine situations where advanced users would create it.